core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableBatchNestedLoopJoin.java

```diff
@@ -0,0 +1,252 @@
+    ParameterExpression corrArg;
+    ParameterExpression corrArgList =
+        Expressions.parameter(Modifier.FINAL,
+            List.class, "corrList");
+
+    // Declare batchSize correlation variables
+    if (!Primitive.is(corrVarType)) {
+      for (int c = 0; c < corrVar.size(); c++) {
+        corrArg =
+            Expressions.parameter(Modifier.FINAL,
+                corrVarType, corrVar.get(c));
+        DeclarationStatement decl = Expressions.declare(
+            Modifier.FINAL,
+            corrArg,
+            Expressions.convert_(
+                Expressions.call(
+                    corrArgList,
+                    BuiltInMethod.LIST_GET.method,
+                    Expressions.constant(c)),
+                corrVarType));
+        corrBlock.add(decl);
+        implementor.registerCorrelVariable(corrVar.get(c), corrArg,
+            corrBlock, leftResult.physType);
+      }
+    } else {
+      for (int c = 0; c < corrVar.size(); c++) {
+        corrArg =
+            Expressions.parameter(Modifier.FINAL,
+                Primitive.box(corrVarType), "$box" + corrVar.get(c));
+        DeclarationStatement decl = Expressions.declare(
+            Modifier.FINAL,
+            corrArg,
+            Expressions.call(
+                corrArgList,
+                BuiltInMethod.LIST_GET.method,
+                Expressions.constant(c)));
+        corrBlock.add(decl);
+        ParameterExpression corrRef =
+            (ParameterExpression) corrBlock.append(corrVar.get(c),
+                Expressions.unbox(corrArg));
+        implementor.registerCorrelVariable(corrVar.get(c), corrRef,
+            corrBlock, leftResult.physType);
+      }
+    }
+    final Result rightResult =
+        implementor.visitChild(this, 1, (EnumerableRel) right, pref);
+
+    corrBlock.add(rightResult.block);
+    for (String c : corrVar) {
+      implementor.clearCorrelVariable(c);
+    }
+
+    final PhysType physType =
+        PhysTypeImpl.of(
+            implementor.getTypeFactory(),
+            getRowType(),
+            pref.prefer(JavaRowFormat.CUSTOM));
+    Expression selector =
+        EnumUtils.joinSelector(
+            joinType, physType,
+            ImmutableList.of(leftResult.physType, rightResult.physType));
+
+    final BlockBuilder builderPredicate = new BlockBuilder();
+    Expression predicate = predicate(implementor,
+        builderPredicate,
+        leftResult.physType,
+        rightResult.physType);
+
+    builder.append(
+        Expressions.call(BuiltInMethod.CORRELATE_BATCH_JOIN.method,
+            Expressions.constant(EnumUtils.toLinq4jJoinType(joinType)),
+            leftExpression,
+            Expressions.lambda(corrBlock.toBlock(), corrArgList),
+            selector,
+            predicate,
+            Expressions.constant(variablesSet.size())));
+    return implementor.result(physType, builder.toBlock());
+  }
+
+  // Predicate "matches" that is gonna be used in correlateBatchJoin to compare two tuples
```

Comment:
Please keep it formal: "gonna" -> "going to" :)

Reply:
Thanks for the correction, but preposition "to" is still missing in the comment.